plugin/trino-hive/src/main/java/io/trino/plugin/hive/ion/IonPageSourceFactory.java

```diff
@@ -147,4 +182,20 @@ public Optional<ReaderPageSource> createPageSource(
             throw new TrinoException(HIVE_CANNOT_OPEN_SPLIT, splitError(e, path, start, length), e);
         }
     }
+
+    private boolean isUnsupportedProperty(Map.Entry<String, String> entry)
+    {
+        String key = entry.getKey();
+        String value = entry.getValue();
+
+        // Check exact matches
+        String propertyDefault = EXACT_PROPERTIES.get(key);
+        if (propertyDefault != null) {
+            return !propertyDefault.equals(value);
+        }
+
+        // Check pattern matches
```

Comment:
This is very much "what", prefer a "why" comment here. I think the insight is that _for now_ we're considering any presence of a "column-specific" property to be non-default.